tools/cmd/golden-test/main.go

```diff
@@ -178,12 +178,16 @@ func ParseDiagnostics(reader io.Reader) (map[Diagnostic]bool, error) {
 
 // WriteDiff writes the summary and the diff (if the base and test are different) between the base
 // and test diagnostics to the writer. If the writer is os.Stdout, it will write the diff in color.
-func WriteDiff(writer io.Writer, branches [2]*BranchResult) {
+func WriteDiff(writer io.Writer, branches [2]*BranchResult, groupingFlag string) {
```

Comment:
same as above: should we just use a `bool` and only convert whenever we have to for better type safety?